src/core/createEventManager.js

```diff
@@ -69,6 +58,17 @@ export default ({
           return consent.awaitConsent();
         })
         .then(() => {
+          // NOTE: this calls onBeforeEventSend callback (if configured)
+          event.finalize(onBeforeEventSend);
+
+          // if the callback returns false, the event should not be sent
+          return event.shouldSend();
+        })
+        .then(shouldSendEvent => {
+          if (shouldSendEvent === false) {
+            return Promise.resolve();
```

Comment:
Do we need any sort of cleanup at this point? I can't think of anything, just thought I'd ask :)

Reply:
@Aaronius - can you spot anything else we need to do here?